pkgs/by-name/me/mesonlsp/package.nix

```diff
@@ -0,0 +1,174 @@
+        hash = "sha256-k883mKwuP35f0WtwX8ybl9uYbvA3y6Vxtv2EJMpZDEs=";
+      };
+
+      sha256 = fetchFromGitHub {
+        owner = "amosnier";
+        repo = "sha-2";
+        rev = "49265c656f9b370da660531db8cc6bf0a2e110a6";
+        hash = "sha256-X9M/ZATYXUiE4oGorPBnsdaKnKaObarnMRh6QEfkBls=";
+      };
+
+      tomlplusplus = fetchFromGitHub {
+        owner = "marzer";
+        repo = "tomlplusplus";
+        rev = "v3.4.0";
+        hash = "sha256-h5tbO0Rv2tZezY58yUbyRVpsfRjY3i+5TPkkxr6La8M=";
+      };
+
+      tree-sitter = fetchFromGitHub {
+        owner = "tree-sitter";
+        repo = "tree-sitter";
+        rev = "v0.20.8";
+        hash = "sha256-278zU5CLNOwphGBUa4cGwjBqRJ87dhHMzFirZB09gYM=";
+      };
+
+      tree-sitter-ini = fetchFromGitHub {
+        owner = "JCWasmx86";
+        repo = "tree-sitter-ini";
+        rev = "20aa563306e9406ac55babb4474521060df90a30";
+        hash = "sha256-1hHjtghBIf7lOPpupT1pUCZQCnzUi4Qt/yHSCdjMhCU=";
+      };
+
+      tree-sitter-meson = fetchFromGitHub {
+        owner = "JCWasmx86";
+        repo = "tree-sitter-meson";
+        rev = "09665faff74548820c10d77dd8738cd76d488572";
+        hash = "sha256-ice2NdK1/U3NylIQDnNCN41rK/G6uqFOX+OeNf3zm18=";
+      };
+    in
+    ''(
+    cd $sourceRoot/subprojects
+
+    cp -R --no-preserve=mode,ownership ${ada} ada
+    cp "packagefiles/ada/meson.build" ada
+
+    cp -R --no-preserve=mode,ownership ${muon} muon
+
+    cp -R --no-preserve=mode,ownership ${sha256} sha256
+    cp "packagefiles/sha256/meson.build" sha256
+
+    cp -R --no-preserve=mode,ownership ${tomlplusplus} tomlplusplus-3.4.0
+
+    cp -R --no-preserve=mode,ownership ${tree-sitter} tree-sitter-0.20.8
+    cp "packagefiles/tree-sitter-0.20.8/meson.build" tree-sitter-0.20.8
+
+    cp -R --no-preserve=mode,ownership ${tree-sitter-ini} tree-sitter-ini
+    cp "packagefiles/tree-sitter-ini/meson.build" tree-sitter-ini
+
+    cp -R --no-preserve=mode,ownership ${tree-sitter-meson} tree-sitter-meson
+    cp "packagefiles/tree-sitter-meson/meson.build" tree-sitter-meson
+  )'';
+
+  postPatch = ''
+    substituteInPlace subprojects/muon/include/compilers.h \
+      --replace-fail 'compiler_language new' 'compiler_language new_'
+
+    substituteInPlace src/liblangserver/langserver.cpp \
+      --replace-fail '"pkg-config"' '"${lib.getExe pkg-config}"'
+
+    substituteInPlace src/libwrap/gitwrap.cpp \
+      --replace-fail '"git"' '"${lib.getExe git}"'
+
+    substituteInPlace src/libwrap/hgwrap.cpp \
+      --replace-fail '"hg"' '"${lib.getExe mercurial}"'
+
+    substituteInPlace src/libwrap/svnwrap.cpp \
+      --replace-fail '"svn"' '"${lib.getExe subversion}"'
+
+    substituteInPlace src/libwrap/wrap.cpp \
+      --replace-fail '"git"' '"${lib.getExe git}"' \
+      --replace-fail '"patch"' '"${lib.getExe patch}"'
+
+    substituteInPlace tests/libutils/processtest.cpp \
+      --replace-fail '"pkg-config"' '"${lib.getExe pkg-config}"'
```

Comment:
We should probably just take svn, hg and git from path to not bloat the closure of the package with unnecessary things.